### ISSUE TITLE
[Bug] sample_polytope step size is correlated with direction when number of samples is small

### ISSUE THREAD

**Issue by sbernasek-albert** [pytorch/botorch#2156]
# 🐛 Bug

In the `botorch.utils.sampling.sample_polytope()` function the same seed is used to generate both the random step sizes and the random step directions.  For very small numbers of steps (e.g. n=1), this causes the the step size and step direction to be strongly correlated, yielding strong sampling bias.

This is irrelevant for those using the `optimize_acqf` procedure that leverages large batches of hit-and-run samples generated in parallel. However, for custom optimization routines that require sequential hit-and-run samples, it may be necessary to call `sample_polytope(..., n=1)`. 

Looping through many such sequential calls yields this correlation between step size (`rands`) and step direction (`Rs`):

![image](https://github.com/pytorch/botorch/assets/150554991/59e8eaa7-5ed6-45e8-ad24-6b7f56911f2b)

A super simple fix would be to add 1 to the seed used to generate a step direction, e.g.  `sample_hypersphere(..., seed=seed+1)` . This eliminates any correlation between step-size and step-direction.

![image](https://github.com/pytorch/botorch/assets/150554991/5fd62b9a-0086-4c1c-91b6-b99c0468fe62)

**Comment by saitcakmak** [comment]
Hi @sbernasek-albert. Apologies for the late response. The proposed fix seems quite reasonable. We'd be happy to accept a PR implementing the fix.

**Comment by Balandat** [comment]
Fixed by #2290
